This file contains information on what filters were applied to the data. 

In [1]:
import pandas as pd

In [2]:
Unfiltered_Merged_06 = pd.read_csv("Unfiltered_MergedData_06.csv")

In [3]:
Unfiltered_Merged_06.columns

Index(['medwid', 'no_lanes', 'lanewid', 'aadt', 'trf_cntl', 'curv_rad',
       'begmp', 'seg_lng', 'endmp', 'access', 'oneway', 'spd_limt', 'med_type',
       'surf_typ', 'cnty_rte', 'rururb', 'rodwycls', 'AccCount',
       'Severity_PDO', 'Severity_Inj_TypeA', 'Severity_Inj_TypeB',
       'Severity_Inj_TypeC', 'Severity_Fatal', 'Severity_Unknown',
       'Weather_Clear', 'Weather_Rain', 'Weather_Snow', 'Weather_Fog',
       'Weather_Hail', 'Weather_CrossWind', 'Weather_Other',
       'Weather_Unknown'],
      dtype='object')

MEDIAN WIDTH AND TYPE

In [4]:
Unfiltered_Merged_06.med_type.value_counts()

0    115156
1     17571
2     11348
5      8768
3      4416
7      1724
4      1710
Name: med_type, dtype: int64

In [5]:
Unfiltered_Merged_06[Unfiltered_Merged_06.med_type!=0].medwid.value_counts()

4      4842
12     4091
64     3732
40     3205
14     2659
       ... 
295       1
540       1
550       1
410       1
475       1
Name: medwid, Length: 148, dtype: int64

In [6]:
Unfiltered_Merged_06[Unfiltered_Merged_06.med_type==0].medwid.sum()

0

In [7]:
print(sum(Unfiltered_Merged_06.med_type.isna()))
print(sum(Unfiltered_Merged_06[Unfiltered_Merged_06.med_type==0].medwid.isna()))
print(sum(Unfiltered_Merged_06[Unfiltered_Merged_06.med_type!=0].medwid.isna()))

0
0
0


It is seen that if the median exists, median width > 0 and if median doesn't exist, median width == 0. Also, median type has 7 (maybe 8) categories. We could do some feature engineering here?

NUMBER OF LANES AND LANE WIDTH

In [8]:
Unfiltered_Merged_06.no_lanes.isna().sum()

0

In [9]:
Unfiltered_Merged_06.no_lanes.value_counts()

2     107948
4      44471
6       4706
3       1752
8        904
5        476
1        175
10        88
12        84
7         61
9         27
0          1
Name: no_lanes, dtype: int64

In [10]:
#One case where the number of lanes = 0 to be removed
Unfiltered_Merged_06[Unfiltered_Merged_06.no_lanes==0]

medwid  no_lanes  lanewid  aadt trf_cntl  curv_rad  begmp  seg_lng  \
63009       0         0        0   175      NaN         0   0.12     0.03   

       endmp  access  ...  Severity_Fatal  Severity_Unknown  Weather_Clear  \
63009   0.15       0  ...               0                 0              0   

       Weather_Rain  Weather_Snow  Weather_Fog  Weather_Hail  \
63009             0             0            0             0   

       Weather_CrossWind  Weather_Other  Weather_Unknown  
63009                  0              0                0  

[1 rows x 32 columns]

In [11]:
Unfiltered_Merged_06[Unfiltered_Merged_06.no_lanes==0].AccCount

63009    0
Name: AccCount, dtype: int64

In [12]:
Unfiltered_Merged_06[Unfiltered_Merged_06.no_lanes!=0].lanewid.value_counts().sort_index()

0      1627
4        11
5        99
6        25
7         8
8        63
9      3710
10     6782
11    25720
12    98800
13    14677
14     3464
15     2165
16     1099
17      468
18      644
19      223
20      423
21      157
22      189
23       72
24      126
25       32
26       63
27       16
28        5
29        1
30       10
31        3
32        7
33        1
36        2
Name: lanewid, dtype: int64

In [13]:
Unfiltered_Merged_06[Unfiltered_Merged_06.lanewid==36]

medwid  no_lanes  lanewid   aadt trf_cntl  curv_rad  begmp  seg_lng  \
19299       0         1       36  10200        9         0  32.65     0.18   
19300       0         1       36  10200      NaN         0  32.78     0.18   

       endmp  access  ...  Severity_Fatal  Severity_Unknown  Weather_Clear  \
19299  32.83       2  ...               0                 0              0   
19300  32.96       2  ...               0                 0              1   

       Weather_Rain  Weather_Snow  Weather_Fog  Weather_Hail  \
19299             0             0            0             0   
19300             0             0            0             0   

       Weather_CrossWind  Weather_Other  Weather_Unknown  
19299                  0              0                0  
19300                  0              0                0  

[2 rows x 32 columns]

In [14]:
Unfiltered_Merged_06[Unfiltered_Merged_06.lanewid==36].cnty_rte

19299    164041
19300    164041
Name: cnty_rte, dtype: int64

The standard lane width is 12 feet 
https://safety.fhwa.dot.gov/geometric/pubs/mitigationstrategies/chapter3/3_lanewidth.cfm

In [15]:
Unfiltered_Merged_06[(Unfiltered_Merged_06.no_lanes!=0)&(Unfiltered_Merged_06.lanewid<12)].shape

(38045, 32)

In [16]:
Unfiltered_Merged_06[(Unfiltered_Merged_06.no_lanes!=0)&(Unfiltered_Merged_06.lanewid<12)][['lanewid','AccCount']]

lanewid  AccCount
73           11         0
74           11         0
75           11         2
76           11         0
77           11         0
...         ...       ...
160610        0         0
160611        0         0
160612        0         0
160613       10         0
160614       10         0

[38045 rows x 2 columns]

In [17]:
Unfiltered_Merged_06[(Unfiltered_Merged_06.no_lanes!=0)&(Unfiltered_Merged_06.lanewid<12)].AccCount

73        0
74        0
75        2
76        0
77        0
         ..
160610    0
160611    0
160612    0
160613    0
160614    0
Name: AccCount, Length: 38045, dtype: int64

In [18]:
Unfiltered_Merged_06[(Unfiltered_Merged_06.no_lanes!=0)&(Unfiltered_Merged_06.lanewid<12)].AccCount.sum()

38730

In [19]:
Unfiltered_Merged_06[(Unfiltered_Merged_06.no_lanes!=0)&(Unfiltered_Merged_06.lanewid<12)].AccCount.max()

91

In [20]:
Unfiltered_Merged_06[(Unfiltered_Merged_06.no_lanes!=0)&(Unfiltered_Merged_06.lanewid<12)].AccCount.min()

0

In [21]:
Unfiltered_Merged_06[(Unfiltered_Merged_06.no_lanes!=0)].AccCount.sum()

197859

In [22]:
Unfiltered_Merged_06[(Unfiltered_Merged_06.no_lanes!=0)&(Unfiltered_Merged_06.lanewid<10)].AccCount.sum()

1694

In [23]:
Unfiltered_Merged_06[(Unfiltered_Merged_06.no_lanes!=0)&(Unfiltered_Merged_06.lanewid<10)].shape

(5543, 32)

Possibly remove all road segments where the lane width < 10. Result in removing 1694 accidents. Also remove all cases where no_lanes = 0

AADT

In [24]:
Unfiltered_Merged_06[Unfiltered_Merged_06.aadt==0]

medwid  no_lanes  lanewid  aadt trf_cntl  curv_rad  begmp  seg_lng  \
1667         0         2        0     0        N         0   0.00     0.01   
1668         0         2        0     0        N         0   0.01     0.23   
1669         0         2        0     0      NaN         0   0.24     0.01   
1681         0         2        0     0        N         0   0.00     0.03   
1682         0         2        0     0      NaN         0   0.03     0.05   
...        ...       ...      ...   ...      ...       ...    ...      ...   
160607       0         2        0     0        9         0   0.25     0.07   
160608       0         2        0     0        9         0   0.32     0.12   
160609       0         2        0     0        9         0   0.44     0.06   
160613       0         2       10     0        N         0   0.00     0.01   
160614       0         2       10     0        9         0   0.01     0.04   

        endmp  access  ...  Severity_Fatal  Severity_Unknown  Weather_Clear  \
1667     0.01       0  ...               0                 0              0   
1668     0.24       0  ...               0                 0              0   
1669     0.25       0  ...               0                 0              0   
1681     0.03       0  ...               0                 0              0   
1682     0.08       0  ...               0                 0              0   
...       ...     ...  ...             ...               ...            ...   
160607   0.32       1  ...               0                 0              1   
160608   0.44       1  ...               0                 0              0   
160609   0.50       1  ...               0                 0              0   
160613   0.01       1  ...               0                 0              0   
160614   0.05       1  ...               0                 0              0   

        Weather_Rain  Weather_Snow  Weather_Fog  Weather_Hail  \
1667               0             0            0             0   
1668               0             0            0             0   
1669               0             0            0             0   
1681               0             0            0             0   
1682               0             0            0             0   
...              ...           ...          ...           ...   
160607             0             0            0             0   
160608             0             0            0             0   
160609             0             0            0             0   
160613             0             0            0             0   
160614             0             0            0             0   

        Weather_CrossWind  Weather_Other  Weather_Unknown  
1667                    0              0                0  
1668                    0              0                0  
1669                    0              0                0  
1681                    0              0                0  
1682                    0              0                0  
...                   ...            ...              ...  
160607                  0              0                0  
160608                  0              0                0  
160609                  0              0                0  
160613                  0              0                0  
160614                  0              0                0  

[1576 rows x 32 columns]

In [25]:
Unfiltered_Merged_06.aadt.value_counts().sort_index().head(10)

0      1576
9       119
10       12
25      136
29      170
50      152
59      176
75      106
79       43
100      98
Name: aadt, dtype: int64

In [26]:
Unfiltered_Merged_06.aadt.isna().sum()

0

In [27]:
Unfiltered_Merged_06[Unfiltered_Merged_06.aadt==0].AccCount.sum()

647

In [28]:
Unfiltered_Merged_06.aadt.max()

332400

In [29]:
Unfiltered_Merged_06.aadt.value_counts().sort_index(ascending=False).head(20)

332400    1
322500    1
322400    3
318800    3
315700    3
314900    2
313300    6
312800    5
311300    1
310700    1
309300    1
308800    1
308500    1
307800    1
307100    1
307000    1
306000    1
305900    1
305500    2
304900    1
Name: aadt, dtype: int64

In [30]:
Unfiltered_Merged_06[Unfiltered_Merged_06.aadt>300000].shape

(57, 32)

In [31]:
Unfiltered_Merged_06[Unfiltered_Merged_06.aadt>300000].AccCount.sum()

4

There are 1576 locations where AADT = 0. 647 accidents are recorded. These should be removed. 

TRAFFIC CONTROL DEVICE

In [32]:
Unfiltered_Merged_06.trf_cntl.isna().sum()

95373

In [33]:
Unfiltered_Merged_06.shape[0]

160693

Nearly 60% of the records have NA for type of traffic control. Should be eliminated

CURVE RADIUS

In [35]:
Unfiltered_Merged_06.curv_rad.isna().sum()

0

In [36]:
Unfiltered_Merged_06.curv_rad.value_counts().sort_index().head(10)

0     157127
15         1
25         2
30         2
35         1
40         1
45         2
48         1
50        89
54         1
Name: curv_rad, dtype: int64

In [37]:
Unfiltered_Merged_06[Unfiltered_Merged_06.curv_rad!=0].shape

(3566, 32)

Curve Radius = 0 means straight segment. Most cases are straight. It should be kept 

SEGMENT LENGTH

In [39]:
Unfiltered_Merged_06.seg_lng.isna().sum()

0

In [40]:
Unfiltered_Merged_06.seg_lng.value_counts().sort_index().head(10)

0.01    22682
0.02    16492
0.03    13841
0.04    14181
0.05    10574
0.06    12912
0.07     8682
0.08     8880
0.09     4982
0.10     4738
Name: seg_lng, dtype: int64

In [41]:
Unfiltered_Merged_06.seg_lng.max()

3.19

In [42]:
Unfiltered_Merged_06.seg_lng.min()

0.01

Segment Length seems ok. It should be kept. 

ROADWAY ACCESS TYPE

In [43]:
Unfiltered_Merged_06.access.isna().sum()

0

In [44]:
Unfiltered_Merged_06.access.value_counts().head(10)

0    129347
2     16842
1     14504
Name: access, dtype: int64

Just three categories. Should be kept. 

ONEWAY

In [45]:
Unfiltered_Merged_06.oneway.isna().sum()

0

In [46]:
Unfiltered_Merged_06.oneway.value_counts().head(10)

2    157740
1      2953
Name: oneway, dtype: int64

Just two categories. Should be kept. 

SPEED LIMIT

In [47]:
Unfiltered_Merged_06.spd_limt.isna().sum()

0

In [48]:
Unfiltered_Merged_06.spd_limt.value_counts()

55    59543
0     25987
30    16295
35    15066
45    13744
65    11329
40    10962
50     5632
25     1850
20      260
10       20
15        5
Name: spd_limt, dtype: int64

In [49]:
Unfiltered_Merged_06[Unfiltered_Merged_06.spd_limt==0].AccCount.sum()

11186

In [50]:
Unfiltered_Merged_06[Unfiltered_Merged_06.spd_limt==0].AccCount.max()

85

In [51]:
Unfiltered_Merged_06[Unfiltered_Merged_06.spd_limt==0].AccCount.min()

0

25987 cases where the speed limit is unknown. There are 11186 accidents. 
1. Should filter out OR
2. Make a guess of the speed limit based on other conditions from data. 

SURFACE TYPE

In [52]:
Unfiltered_Merged_06.surf_typ.isna().sum()

0

In [53]:
Unfiltered_Merged_06.surf_typ.value_counts()

600    53155
610    37760
620    17135
630    14742
640     6373
550     5611
560     4649
730     3803
740     3560
720     3024
700     2233
710     1834
500     1475
650     1251
725     1242
300      629
976      492
965      469
967      311
200      235
400      111
410      106
975       91
800       67
625       50
10        45
963       40
530       27
790       22
973       21
775       21
20        18
987       15
520       12
954       12
210       10
972       10
792        9
956        7
0          4
957        4
964        3
110        2
971        2
974        1
Name: surf_typ, dtype: int64

Large number of categories. We should concolidate categories based on commonalities. 

RURAL OR URBAN

In [54]:
Unfiltered_Merged_06.rururb.isna().sum()

0

In [55]:
Unfiltered_Merged_06.rururb.value_counts()

1    89619
2    71074
Name: rururb, dtype: int64

Should be kept

ROADWAY CLASS

In [56]:
Unfiltered_Merged_06.rodwycls.isna().sum()

0

In [57]:
Unfiltered_Merged_06.rodwycls.value_counts()

8     77685
3     27641
4     18969
5     12698
1      8558
6      7440
99     3282
9      3169
10      519
7       467
2       265
Name: rodwycls, dtype: int64

11 categories. May consider consolidating categories. Should be kept. 

In [61]:
Filtered_Merged_06 = Unfiltered_Merged_06[(Unfiltered_Merged_06.no_lanes!=0)&(Unfiltered_Merged_06.lanewid>=10)&(Unfiltered_Merged_06.aadt>0)]

In [63]:
Filtered_Merged_06['Year'] = '06'

/Users/jacob/anaconda/envs/py3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [64]:
Filtered_Merged_06

medwid  no_lanes  lanewid   aadt trf_cntl  curv_rad   begmp  seg_lng  \
0            0         2       12   8600        N         0    0.00     0.01   
1            0         2       12   8600      NaN         0    0.01     0.12   
2            0         2       12   8600        N         0    0.13     0.03   
3            0         2       12   8600      NaN         0    0.16     0.02   
4            0         2       18   8600      NaN         0    0.18     0.03   
...        ...       ...      ...    ...      ...       ...     ...      ...   
160688      64         4       12  22500      NaN         0  116.07     0.26   
160689      64         4       12  22500      NaN         0  116.33     0.29   
160690      64         4       12  22500      NaN         0  116.62     0.65   
160691      64         4       12  22500      NaN         0  117.27     0.97   
160692      64         4       12  22500      NaN         0  118.24     0.36   

         endmp  access  ...  Severity_Unknown  Weather_Clear  Weather_Rain  \
0         0.01       0  ...                 0              0             0   
1         0.13       0  ...                 0              0             0   
2         0.16       0  ...                 0              0             0   
3         0.18       0  ...                 0              0             0   
4         0.21       0  ...                 0              2             0   
...        ...     ...  ...               ...            ...           ...   
160688  116.33       2  ...                 0              2             0   
160689  116.62       2  ...                 0              0             1   
160690  117.27       2  ...                 0              3             1   
160691  118.24       2  ...                 0              1             0   
160692  118.60       2  ...                 0              0             0   

        Weather_Snow  Weather_Fog  Weather_Hail  Weather_CrossWind  \
0                  0            0             0                  0   
1                  0            0             0                  0   
2                  0            0             0                  0   
3                  0            0             0                  0   
4                  0            0             0                  0   
...              ...          ...           ...                ...   
160688             0            0             0                  0   
160689             1            0             0                  0   
160690             1            0             0                  0   
160691             0            0             0                  0   
160692             1            0             0                  0   

        Weather_Other  Weather_Unknown  Year  
0                   0                0    06  
1                   0                0    06  
2                   0                0    06  
3                   0                0    06  
4                   0                0    06  
...               ...              ...   ...  
160688              0                0    06  
160689              0                0    06  
160690              0                0    06  
160691              0                0    06  
160692              0                0    06  

[154727 rows x 33 columns]

In [66]:
Filtered_Merged_06.to_csv('Filtered_Merged_06.csv',index=False)